boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -200,11 +205,15 @@ ifeq ($(CONFIG_SIM_M32),y)
   HOSTLDFLAGS += -m32
 endif
 
-ifeq ($(CONFIG_SIM_SANITIZE),y)
-  CCLINKFLAGS += -fsanitize=address -fsanitize=undefined -fno-omit-frame-pointer
+ifeq ($(CONFIG_SIM_ASAN),y)
+  CCLINKFLAGS += -fsanitize=address -fno-omit-frame-pointer
 else ifeq ($(CONFIG_MM_KASAN),y)
   CCLINKFLAGS += -fsanitize=kernel-address
 endif
 
+ifeq ($(CONFIG_SIM_UBSAN),y)
+  CCLINKFLAGS += -fsanitize=undefined
```

Comment:
According to [clang documentation](https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#stack-traces-and-report-symbolization), `-fno-omit-frame-pointer` also contributes to UBSan.
Consider adding it via a hidden Kconfig option that is selected when either ASan or UBSan is selected.

Reply:
It's too complex to add a new Kconfig to handle it. It's fine to add -fno-omit-frame-pointer twice. Please take a look.